onc-healthit/lantern-back-end
# Netstats Querier processes endpoints from queue

**This PR Covers**:
- The netstats querier processing the endpoints from a queue instead of a file
- Moving the queueworkers out of the capabilityquerier
- Renaming the queueworkers to workers and making it more general

**To Test**:
- Add endpoints to the database (`make populatedb` or another alternative)
- Send the endpoints to the queues: in `endpointmanager/cmd`, run `go run sendendpoints/main.go`
- Log in to the rabbitmq page on `localhost:15672` and see the messages on the `endpoints-to-netstats` queue being processed
- You should also see log messages from `prometheus_postgresql_adapter_1` in the `make run` window with the endpoints.
- There should also be values in the `metrics_values` table.

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-162
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
